src/index.ts

```diff
@@ -19,17 +19,24 @@ async function deliver(url: string, secret: string, payload: string): Promise<Ax
     contextUrl = `https://github.com/${repo.owner}/${repo.repo}/actions/runs/${GITHUB_RUN_ID}`;
     core.info(`GitHub Context ${contextUrl}`);
   }
-
-  const headSha = workFlowPaylod?.pull_request?.head?.sha ?? sha;
+  // when the event is run with a workflow_run event
+  const targetWorkflowRun = workFlowPaylod?.workflow_run
+  core.info(`Target workflow run: ${JSON.stringify(targetWorkflowRun)}`)
+  const headSha = workFlowPaylod?.pull_request?.head?.sha ?? targetWorkflowRun?.head_sha ?? sha;
   const sender = workFlowPaylod?.sender?.login;
+  let refFromTargetWorkflow: string | null = null;
+  if (targetWorkflowRun?.head_branch != null) {
```

Comment:
You can just rewrite this as `if (targetWorkflowRun?.head_branch) { ... }`

Reply:
WDYT about keeping the explicit != null since it's pretty surprising what js sees as truthy / falsey?

Reply:
i changed it to be consistent because this file already does it above for `GITHUB_RUN_ID`.